client/src/components/TimeTable.jsx

```diff
@@ -0,0 +1,201 @@
+                    Days: "Tuesday",
+                    Hours:"1",
+                    Section: "T11"
+                },
+                {
+                    Name: "DSA",
+                    ID: "CS F211",
+                    Days: "Thursday",
+                    Hours:"1",
+                    Section: "T3"
+                },
+                
+                {
+                    Name: "DSA",
+                    ID: "CS F211",
+                    Days: "Tuesday Thursday Saturday",
+                    Hours:"5 6 7 8",
+                    Section: "L1"
+                },
+                
+            ]
+        }
+        this.populateTimetable.bind(this);
+        this.gridArray = this.populateTimetable(this.state.courses); 
+    }
+    populateTimetable(coursesAdded)
+    {
+        var gridList = [
+            ["Time",1,2,3,4,5,6,7,8,9,10],
+            ["Monday",0,0,0,0,0,0,0,0,0,0],
+            ["Tuesday",0,0,0,0,0,0,0,0,0,0],
+            ["Wednesday",0,0,0,0,0,0,0,0,0,0],
+            ["Thursday",0,0,0,0,0,0,0,0,0,0],
+            ["Friday",0,0,0,0,0,0,0,0,0,0],
+            ["Saturday",0,0,0,0,0,0,0,0,0,0]
+        ];
+       
+        var Map = {};
+         Map['Monday'] = 1;
+         //Map[1] = 'Monday';
+         Map['Tuesday'] = 2;
+         //Map[2] = 'Tuesday';
+         Map['Wednesday'] = 3;
+         //Map[3] = 'Wednesday';
+         Map['Thursday'] = 4;
+         //Map[4] = 'Thursday';
+         Map['Friday'] = 5;
+         //Map[5] = 'Friday';
+         Map['Saturday'] = 6;
+         //Map[6] = 'Saturday';
+            
+            //let each course have a day list, and a hours list, both space separated strings
+            for(var course of coursesAdded)
+            {    
+                
+                var listOfDays = Array.from(course.Days.split(' '));
+                var listOfHours = (Array.from(course.Hours.split(' ')));
+                for(var i = 0; i< listOfHours.length; i++)
+                {
+                    listOfHours[i] = +listOfHours[i];
+                }
+                for(i of listOfDays)
+                {
+                    var divStyle = {};
+                    if(listOfHours.length > 1)
+                    {
+                         divStyle = {
+                           gridRowStart: `${listOfHours[0]+1}`,
+                           gridColumnStart: `${Map[i]+1}`,
+                           gridCoulmnEnd: `${Map[i]+2}`,
+                           gridRowEnd: `span ${listOfHours[listOfHours.length-1]-listOfHours[0]+1}`
+                        }
+                    }
+                    let str = <>
+                                <div className="gridItem" style={divStyle}>
+                                    {course.Name}<br></br>
+                                    {course.ID}<br></br>
+                                    {course.Section}<br></br>
+                                </div>
+                            </>;
```

Comment:
`<>`, `</>` are meant for when there isn't a single root iirc.
I don't think it is needed here

Reply:
Lite, but will change